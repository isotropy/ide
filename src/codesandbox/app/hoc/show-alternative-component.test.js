import React from 'react';
import testRender from 'codesandbox/app/utils/test/render';

import showAlternativeComponent from './show-alternative-component';

describe('hoc', () => {
  describe('show-alternative-component', () => {
    const LoadingComponent = () => <div>Loading</div>;
    const AppComponent = () => <div>App</div>;
    it("renders loading component if prop doesn't exist", () => {
      const ComposedComponent = showAlternativeComponent(LoadingComponent, [
        'field',
      ])(AppComponent);

      testRender(<ComposedComponent />);
    });

    it('renders app if prop exists', () => {
      const ComposedComponent = showAlternativeComponent(LoadingComponent, [
        'field',
      ])(AppComponent);

      testRender(<ComposedComponent field="test" />);
    });

    it('renders app if prop exists and is falsy', () => {
      const ComposedComponent = showAlternativeComponent(LoadingComponent, [
        'field',
      ])(AppComponent);

      testRender(<ComposedComponent field={false} />);
    });

    it('only renders app if all prop exists and is falsy', () => {
      const ComposedComponent = showAlternativeComponent(LoadingComponent, [
        'field',
        'field2',
      ])(AppComponent);

      testRender(<ComposedComponent field={false} />);
      testRender(<ComposedComponent field={false} field2={false} />);
    });
  });
});
