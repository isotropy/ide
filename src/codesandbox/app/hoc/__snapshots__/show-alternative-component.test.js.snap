// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hoc show-alternative-component only renders app if all prop exists and is falsy 1`] = `
<div>
  Loading
</div>
`;

exports[`hoc show-alternative-component only renders app if all prop exists and is falsy 2`] = `
<div>
  App
</div>
`;

exports[`hoc show-alternative-component renders app if prop exists 1`] = `
<div>
  App
</div>
`;

exports[`hoc show-alternative-component renders app if prop exists and is falsy 1`] = `
<div>
  App
</div>
`;

exports[`hoc show-alternative-component renders loading component if prop doesn't exist 1`] = `
<div>
  Loading
</div>
`;
